Mathlib/Algebra/Algebra/Operations.lean

```diff
@@ -353,6 +355,25 @@ theorem comap_op_mul (M N : Submodule R Aᵐᵒᵖ) :
         comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) M := by
   simp_rw [comap_equiv_eq_map_symm, map_unop_mul]
 
+section
+variable {B : Type*} [Semiring B] [Algebra R B] [Module A B] [SMulCommClass A R B]
+
+instance [IsScalarTower A B B] : IsScalarTower A (Submodule R B) (Submodule R B) where
```

Comment:
I'm pretty sure you don't need `Algebra R A` here? That is, you've taken more assumptions than you need by reusing `A` for the wrong thing. Can you write this `IsScalarTower S (Submodule R A) (Submodule R A)` for a new variable `S`?

Reply:
`variable {S : Type*} [Monoid S] [DistribMulAction S A] [SMulCommClass S R A]` seems to suffice

Reply:
(ok, maybe `S` is a bad name after all, but you could pick a greek letter or a subscripted `R`)

Reply:
Oh, I meant for `A` to be a `R`-algebra rather than the other way around